### ISSUE TITLE
Parser diverges instead of failing on ill-formed raw blocks

### ISSUE THREAD

**Issue by thumphries** [wycats/handlebars.js#1156]
Hello,

```
{{{{regular}}}} {{{{ expressions }}}} {{{{/expressions}}}} {{{{/  regular}}}}
```

Handlebars will diverge and eventually run out of memory on the above inputs. The culprit is the space before `regular` in the outer raw close. In the lexer, this seems like input that should be reasonably quickly rejected by https://github.com/wycats/handlebars.js/blob/master/src/handlebars.l#L54, so that makes me think the greedy match immediately below that is also culpable.

Note that whenever the closing raw block is invalid after a correct `{{{{/` it seems to do the same thing:

```
{{{{regular}}}} {{{{ expressions }}}} {{{{/expressions}}}}  {{{{/awoeifjawoiefjaoiwejfoaiwejfio}}}
```

```
{{{{regular}}}} {{{{ expressions }}}} {{{{/expressions}}}}  {{{{/awoeifjawoiefjaoiwejfoaiwe
```

```
{{{{regular}}}} {{{{ expressions }}}} {{{{/expressions}}}}  {{{{/mary had a little lamb
```

**Comment by thumphries** [comment]
The nesting also appears to be irrelevant, i.e. it still crashes on

```
{{{{regular}}}} {{{{/ regular}}}}
```

```
==== JS stack trace =========================================

Security context: 0x1bc374d44a49 <JS Object>
    1: anonymous [/usr/local/lib/node_modules/handlebars/dist/cjs/handlebars/compiler/parser.js:~11] [pc=0x2e09dbfa2e8a] (this=0x35dfe0dfa821 <an Object with map 0x10a715869621>,yytext=0x1bc374dfdf89 <String[0]: >,yyleng=0,yylineno=0,yy=0x3a967ba8cdc9 <an Object with map 0x10a715869411>,yystate=10,$$=0x1bc374dfdf49 <JS Array[5]>,_$=0x1bc374dfdf69 <JS Array[5]>)
    2: parse [/usr/local/lib/node...

FATAL ERROR: CALL_AND_RETRY_LAST Allocation failed - process out of memory
```

**Comment by damncabbage** [comment]
Confirmed here with this jsfiddle: https://jsfiddle.net/9D88g/50/

<img width="831" alt="screen shot 2016-02-18 at 6 33 22 pm" src="https://cloud.githubusercontent.com/assets/133028/13136532/826b0de2-d66e-11e5-8c09-65408423db87.png">

(Don't click the link unless you want the browser to chew up a bunch of memory.)

**Comment by mootari** [comment]
Also running out of memory with:
```
{{{{raw}}}} {{{{/raw }}}}
```

**Comment by nknapp** [comment]
Seems to be fixed with #1578
